servers/physics_3d/space_3d_sw.cpp

```diff
@@ -549,7 +549,7 @@ int Space3DSW::_cull_aabb_for_body(Body3DSW *p_body, const AABB &p_aabb) {
 			keep = false;
 		} else if (intersection_query_results[i]->get_type() == CollisionObject3DSW::TYPE_SOFT_BODY) {
 			keep = false;
-		} else if ((static_cast<Body3DSW *>(intersection_query_results[i])->test_collision_mask(p_body)) == 0) {
+		} else if (!p_body->layer_in_mask(static_cast<Body3DSW *>(intersection_query_results[i]))) {
```

Comment:
Actually after doing more tests I realized this layer check is inverted, a kinematic body should collide with objects it scans in its own mask, not objects which have its own layer in their mask. I'll make a fix in another PR.

Reply:
This is the correct way around. A `KinematicBody` should only interact with other `Bodies` that have a matching mask.

Reply:
The way you implemented things, walls would define what they stop instead of moving bodies what they collide with. That would be a nightmare for users to setup.

Reply:
You're right. Its mask is the layers it sees. So it is the other way around.

Reply:
It probably makes more sense then to revert the logic (and the name) of `layer_in_mask` to `mask_has_layer`.